src/Assignments/ProblemEditorAssetUploadButton.tsx

```diff
@@ -0,0 +1,229 @@
+                                p={1}
+                                flexGrow={1}
+                            >
+                                <TextField
+                                    aria-label="Upload file name"
+                                    label='Upload file name'
+                                    variant='outlined'
+                                    type='text'
+                                    value={file?.name ?? ''}
+                                    disabled={true}
+                                    fullWidth={true}
+                                />
+                            </Box>
+                            {/* This: {...getRootProps()} causes the first button to flicker, need to investigate further */}
+                            {/* <Box
+                                p={1}
+                            >
+                                <div
+                                    style={{
+                                        position: 'relative',
+                                        height: '100%'
+                                    }}
+                                    {...getRootProps()}
+                                >
+                                    <Button
+                                        variant="outlined"
+                                        style={{
+                                            height: '100%'
+                                        }}
+                                    >
+                                        Change PG Asset
+                                    </Button>
+                                    {isDragActive && (
+                                        <div style={{
+                                            position: 'absolute',
+                                            left: 0,
+                                            top: 0,
+                                            right: 0,
+                                            bottom: 0,
+                                            border: '5px dashed lightblue',
+                                            borderRadius: '3px',
+                                            textAlign: 'center',
+                                            zIndex: 2,
+                                            backgroundColor: 'white',
+                                            opacity: 0.9,
+                                        }}
+                                        >
+                                            <div style={{ position: 'relative', margin: '0 auto', top: '15%', fontSize: '1em' }}>
+                                                <FaFileUpload style={{ position: 'relative', margin: '0 auto', top: '15%', display: 'block', fontSize: '1em' }} />
+                                            </div>
+                                        </div>
+                                    )}
+                                </div>
+                            </Box> */}
```

Comment:
this seems to be duplicate code of 88-122?

Reply:
It does seem that way but the idea is that you can change the item that you want to upload once in the modal, the problem I was facing was that the dropzone misbehaved on the first button when used twice